### ISSUE TITLE
assert(0) as ternary RHS => crash

### ISSUE THREAD

**Issue by maxhaton** [ldc-developers/ldc#3885]
```d
int front(int param)
{
    return param ? 1 : assert(0);
}
```
Also causes an assertion failure in dmd (https://issues.dlang.org/show_bug.cgi?id=22587). Is it even valid code?

**Comment by MoonlightSentinel** [comment]
>  Is it even valid code?

Yes, see the [DIP](https://github.com/dlang/DIPs/blob/master/DIPs/accepted/DIP1034.md#interaction-with-other-language-features)

**Comment by maxhaton** [comment]
> > Is it even valid code?
> 
> Yes, see the [DIP](https://github.com/dlang/DIPs/blob/master/DIPs/accepted/DIP1034.md#interaction-with-other-language-features)

Good good, was just asking to be idiot-checked.
